src/richie/apps/search/defaults.py

```diff
@@ -0,0 +1,54 @@
+from django.conf import settings
+from django.forms import ChoiceField
+
+import arrow
+
+
+# Define our aggregations names, for our ES query, which will match with the field
+# names on the objects & the facets we return on the API response
+RESOURCE_FACETS = getattr(
+    settings, "RICHIE_SEARCH_RESOURCE_FACETS", ["organizations", "subjects"]
+)
+
+FILTERS_HARDCODED_DEFAULT = {
+    "availability": {
+        "field": ChoiceField(
+            choices=[("coming_soon", ""), ("current", ""), ("open", "")],
+            required=False
+        ),
+        "choices": {
+            "coming_soon": [
+                {
+                    "range": {
+                        "start_date": {
+                            "gte": arrow.utcnow().datetime,
+                            "lte": arrow.utcnow().shift(weeks=+12).datetime,
+                        }
+                    }
+                }
+            ],
+            "current": [
+                {"range": {"start_date": {"lte": arrow.utcnow().datetime}}},
+                {"range": {"end_date": {"gte": arrow.utcnow().datetime}}},
+                {"range": {"enrollment_start_date": {"lte": arrow.utcnow().datetime}}},
+                {"range": {"enrollment_end_date": {"gte": arrow.utcnow().datetime}}},
+            ],
+            "open": [
+                {"range": {"start_date": {"lte": arrow.utcnow().datetime}}},
+                {"range": {"end_date": {"gte": arrow.utcnow().datetime}}},
+            ],
+        },
+    },
+    "language": {
+        "field": ChoiceField(choices=[("en", ""), ("fr", "")], required=False),
```

Comment:
I think you should just pass the class ChoiceField or MultipleChoiceField. The object you pass for choices is akward and a useless duplicate. A structure that would maybe be more self explanatory : 
```
FILTERS_HARDCODED_DEFAULT = {
    availability : [
        {
            name :  "coming_soon",
            human_name :  "Coming soon",
            query_fragment :  ...,
            field_class :  MultipleChoiceField
        },
        {...}
    ]
}
```

Reply:
I don't think I understand your proposal.

The `ChoiceField` key is on `"language"` whose value can be either `"en"` or `"fr"`.

As far as I can tell there is no point in a ChoiceField if I don't specify the actual choices it allows. That would just be a `CharField` that does not check the actual content of the values...


Also in all the cases we want it's a `ChoiceField` but it might as well be something else entirely?

Reply:
my point was that in the __init__ of your form you could construct the choices and avoid this duplicate. Something like:
```python
self.choices = [(f["name"], f["human_name"]) for f in FILTERS_HARDCODED_DEFAULT["availability"]]
```

Reply:
yes, that's what I understood after reading it the 2nd time and what I actually did this morning. Thanks :)